### ISSUE TITLE
No module named 'tensorflow.python.keras.preprocessing'

### ISSUE THREAD

**Issue by Jwenyi** [DataCanvasIO/Cooka#9]
Hi,
I just installed Cooka by "pip install cooka"; however, the terminal reports `ModuleNotFoundError: No module named 'tensorflow.python.keras.preprocessing'` when starting cooka web server.
It might appear to be a versioning problem?

**Comment by oaksharks** [comment]
Hi @Jwenyi ,  thanks for your report.
It looks like the version of Tensorflow is too new, you can downgrade Tensorflow to `tensorflow>=2.0, <2.5` and try again.
Here is my local environment for your reference:

```
absl-py==0.12.0
argon2-cffi==20.1.0
astunparse==1.6.3
async-generator==1.10
attrs==20.3.0
backcall==0.2.0
bleach==3.3.0
cachetools==4.2.1
catboost==0.25.1
category-encoders==2.2.2
certifi==2020.12.5
cffi==1.14.5
chardet==4.0.0
click==7.1.2
cloudpickle==1.6.0
contextvars==2.4
cooka==0.1.3
cycler==0.10.0
dask==2021.3.0
dask-glm==0.2.0
dask-ml==1.8.0
dask-xgboost==0.1.11
decorator==4.4.2
deeptables==0.1.13
defusedxml==0.7.1
distributed==2021.3.0
eli5==0.11.0
entrypoints==0.3
featuretools==0.23.3
flatbuffers==1.12
fsspec==0.9.0
gast==0.3.3
google-auth==1.28.0
google-auth-oauthlib==0.4.4
google-pasta==0.2.0
graphviz==0.16
greenlet==1.0.0
grpcio==1.32.0
h5py==2.10.0
HeapDict==1.0.1
hypergbm==0.2.2
hypernets==0.1.8
idna==2.10
imageio==2.9.0
imbalanced-learn==0.8.0
imblearn==0.0
immutables==0.15
importlib-metadata==3.10.0
ipykernel==5.5.3
ipython==7.16.1
ipython-genutils==0.2.0
jedi==0.18.0
Jinja2==2.11.3
joblib==1.0.1
json5==0.9.5
jsonschema==3.2.0
jupyter-client==6.1.12
jupyter-core==4.7.1
jupyterlab==2.0.0
jupyterlab-pygments==0.1.2
jupyterlab-server==1.0.9
Keras-Preprocessing==1.1.2
kiwisolver==1.3.1
lightgbm==3.2.0
llvmlite==0.36.0
locket==0.2.1
Markdown==3.3.4
MarkupSafe==1.1.1
matplotlib==3.3.4
mistune==0.8.4
msgpack==1.0.2
multipledispatch==0.6.0
nbclient==0.5.3
nbconvert==6.0.7
nbformat==5.1.3
nest-asyncio==1.5.1
networkx==2.5.1
notebook==6.3.0
numba==0.53.1
numexpr==2.7.3
numpy==1.19.5
oauthlib==3.1.0
opt-einsum==3.3.0
packaging==20.9
pandas==1.1.5
pandocfilters==1.4.3
parso==0.8.2
partd==1.2.0
patsy==0.5.1
pexpect==4.8.0
pickleshare==0.7.5
Pillow==8.2.0
plotly==4.14.3
prometheus-client==0.10.0
prompt-toolkit==3.0.18
protobuf==3.15.8
psutil==5.8.0
ptyprocess==0.7.0
pyaml==20.4.0
pyarrow==2.0.0
pyasn1==0.4.8
pyasn1-modules==0.2.8
pycparser==2.20
Pygments==2.8.1
pyparsing==2.4.7
pyrsistent==0.17.3
python-dateutil==2.8.1
pytz==2021.1
PyWavelets==1.1.1
PyYAML==5.4.1
pyzmq==22.0.3
requests==2.25.1
requests-oauthlib==1.3.0
retrying==1.3.3
rsa==4.7.2
scikit-image==0.17.2
scikit-learn==0.24.1
scikit-optimize==0.7.1
scipy==1.5.4
seaborn==0.11.0
Send2Trash==1.5.0
shap==0.28.5
six==1.15.0
sortedcontainers==2.3.0
SQLAlchemy==1.4.6
statsmodels==0.12.2
supervisor==4.2.2
tables==3.6.1
tabular-toolbox==0.1.4
tabulate==0.8.9
tblib==1.7.0
tensorboard==2.4.1
tensorboard-plugin-wit==1.8.0
tensorflow==2.4.1
tensorflow-estimator==2.4.0
termcolor==1.1.0
terminado==0.9.4
testpath==0.4.4
threadpoolctl==2.1.0
tifffile==2020.9.3
toolz==0.11.1
tornado==6.0.4
tqdm==4.60.0
traitlets==4.3.3
typing-extensions==3.7.4.3
urllib3==1.26.4
wcwidth==0.2.5
webencodings==0.5.1
Werkzeug==1.0.1
wrapt==1.12.1
xgboost==0.90
zict==2.0.0
zipp==3.4.1
```

**Comment by Jwenyi** [comment]
Thanks @oaksharks , Cooka now works well locally :D

And I have another question that how should I cite Cooka, DeepTable in my papers?

**Comment by oaksharks** [comment]
Hi Jwenyi, 

Welcome to use Cooka/DeepTables for research ! We have added **Citation** section in the readme.md:

https://github.com/DataCanvasIO/Cooka/blob/2afccaf01122428493afe8991d6079f42a20a1e9/README.md?plain=1#L116-L129
